src/executor/host.rs

```diff
@@ -1074,39 +1070,32 @@ impl ReadyToRun {
             HostFunction::ext_storage_child_root_version_1 => host_fn_not_implemented!(),
             HostFunction::ext_storage_child_next_key_version_1 => host_fn_not_implemented!(),
             HostFunction::ext_storage_start_transaction_version_1 => {
-                if self.inner.within_storage_transaction {
-                    return HostVm::Error {
-                        error: Error::NestedTransaction,
-                        prototype: self.inner.into_prototype(),
-                    };
-                }
-
-                self.inner.within_storage_transaction = true;
+                self.inner.storage_transaction_depth += 1;
```

Comment:
```suggestion
                // TODO: a maximum depth is important in order to prevent a malicious runtime from crashing the client, but the depth needs to be the same as in Substrate; figure out
                self.inner.storage_transaction_depth += 1;
```

I don't think it's important to enforce a maximum depth right now, but I think we eventually should. Since it's a non-trivial question, I'd go for just a todo.

Reply:
There is a limit enforced by Substrate. I can add it here https://github.com/paritytech/substrate/blob/e8a4408c53ae2a1a0ef6cb2f7b603887a231c16f/frame/support/src/storage/transactional.rs#L37

Reply:
But that's enforced at runtime level not client level so I guess I shouldn't enforce it